plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitSource.java

```diff
@@ -91,7 +92,12 @@ public HudiSplitSource(
                 queue,
                 new BoundedExecutor(executor, getSplitGeneratorParallelism(session)),
                 createSplitWeightProvider(session),
-                partitions);
+                partitions,
+                throwable -> {
+                    trinoException.compareAndSet(null, new TrinoException(GENERIC_INTERNAL_ERROR,
```

Comment:
Change looks good. Is it possible to test the fix by forcing an exception and check through timeouts that the query is not hanging?

Reply:
@codope  added the test case for the fix. Please review